<?xml version="1.0"?>

<launch>

<rosparam param="/uwb_localization_node0/ancsPos"> [-3.713, 4.534, -1.945, 3.031, 2.979, -0.6718, 3.72, -3.518, -2.0, -2.985, -3.011, -0.6213]</rosparam>
<rosparam param="/uwb_localization_node0/ancsPos"> [-5, 7.5, -1.95, 5, 7.5, -0.75, 5, -7.5, -1.95, -5, -7.5, -0.60]</rosparam>
<rosparam param="/uwb_localization_node0/ancsId"> [100, 101, 102, 103]</rosparam>
<rosparam param="/uwb_localization_node0/mobsId"> [200, 201, 202, 203]</rosparam>
<rosparam param="/uwb_localization_node0/ekfCovXyZ"> [5, 12.5]</rosparam>
<rosparam param="/uwb_localization_node0/albega"> [1.001966856066302, 0.108814649293333, 200]</rosparam>
<rosparam param="/node_pos"> [ 4.534, -3.713, 1.945, 2.979, 3.031,  0.6718, -3.518,  3.72, 2.0, -3.011 , -2.985, 0.6213] </rosparam>

 <!-- rcmEkfLocalization node -->
  <node respawn="true" pkg="uwb_localization" type="uwb_localization_node" name="uwb_localization_node0" output="screen">
            <!-- configuration for the node -->
	<param name="rcmLocalizationRate" type="int" value="10"/>
	<param name="reportRate" type="int" value="20"/>
	<param name="restEnable" type="bool" value="true"/>
	<param name="ignrTimeoutUwbInit" type="bool" value="true"/>
	<param name="ignrFailedMpuInit" type="bool" value="true"/>
	<param name="publishPose" type="bool" value="false"/>
	<param name="enableVicon" type="bool" value="true"/>

            <!-- configuration for the least square localization -->
	<param name="TRILAT_TIMES" type="int" value="20"/>
	<param name="timeWindow" type="int" value="250"/>	<!-- timeWindow in millisecond -->
	<param name="errThres" type="double" value="0.01"/>
	<param name="maxIterTimes" type="int" value="40"/>

            <!-- configuration for the p4xx -->
	<param name="p4xxInterfaceChoice" type="int" value="1"/>
	<param name="p4xxSerialPort" type="string" value="204" />
	<param name="mode" type="int" value="0" />
	<param name="autoConfig" type="bool" value="false" />

            <!-- configuration for fcc 
	<param name="fccSerialPort" type="string" value="/dev/serial/by-id/usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0" />
	<param name="fccBaudrate" type="int" value="921600" /> -->
	
	        <!-- configuration for comm. health -->
	<param name="logEnable" type="bool" value="true" />
	<param name="logInput" type="bool" value="true" />
	<param name="logOutput" type="bool" value="true" />
	<param name="syncFCC" type="bool" value="true" />
	<param name="attendanceCycle" type="int" value="5" /> <!-- each cycle ~ 100 ms -->

            <!-- configuration for the ekf -->
	<param name="enableCalib" type="bool" value="false"/>
	<!-- <param name="useImu" type="bool" value="false"/>-->
	<param name="medianFilterSize" type="double" value="5.0"/>
	<param name="uwbAcceptance" type="double" value="1"/>
	<param name="fuseBaro" type="bool" value="false"/>
	<param name="ekfRescueEnable" type="bool" value="false"/>
	<param name="ekfRescueTime" type="double" value="1.0"/>
	<param name="outlierThres" type="int" value="10"/>

            <!-- configuration for the imu 
	<param name="imuSettleTime" type="double" value="1.0"/>		
	<param name="imu_autodetect" type="bool" value="true"/>-->
	
	<param name="i2c_bus" value="1" />
	
	<!-- gyro_accel_sample_rate: 5 - 1000 Hz -->
	<param name="gyro_accel_sample_rate" value="200" />

	<!-- compass_sample_rate: 1 - 100 Hz -->
	<param name="compass_sample_rate" value="100" />	

  </node>
  
<!--<include file="$(find viconXbee)/launch/viconXbee.launch"/> -->
<!--include file="$(find viconXbee)/launch/vicon_local_tf.launch"/-->

   <!-- dummy publisher node -->
  <!--node respawn="true" pkg="dummy_pub" type="dummy_pub_node" name="dummy_pub_node0" output="screen" launch-prefix="taskset -c 2">
  </node-->
  
</launch>
